cyder/base/fields.py

```diff
@@ -51,9 +51,9 @@ def to_python(self, value):
         value = super(MacAddrField, self).to_python(value)
 
         if value:
-            value = value.lower().replace(':', '').replace('-', '')
+            raw = value.lower().replace(':', '').replace('-', '')
             value = reduce(lambda x,y: x + ':' + y,
-                           (value[i:i+2] for i in xrange(0, 12, 2)))
+                           (raw[i:i+2] for i in xrange(0, 12, 2))) + raw[12:]
```

Comment:
I figure that if there's anything to append, we know it's going to return an invalid mac regardless, so it might as well be obvious that we're returning an invalid mac. But if you think that this implementation is more appropriate for logging or whatever reason, I'm willing to let it through.